lms/static/js/instructor_dashboard/data_download_2.js

```diff
@@ -0,0 +1,257 @@
+            };
+
+            this.clear_display();
+
+            /**
+             * Show and hide selected tab data
+             */
+            this.$tabSwitch.click(function(event) {
+                var selectedSection = '#' + $(this).attr('data-section');
+                event.preventDefault();
+                $('.data-download-nav .btn-link').removeClass('active-section');
+                $('section.tab-data').hide();
+                $(selectedSection).show();
+                $(this).addClass('active-section');
+
+                $(this).find('select').trigger('change');
+                dataDownloadObj.$selectedSection = $(selectedSection);
+
+                dataDownloadObj.clear_display();
+            });
+
+            this.$tabSwitch.first().click();
+
+            /**
+             * on change of report select update show and hide related descriptions
+             */
+            this.$report_type_selector.change(function() {
+                var selectedOption = dataDownloadObj.$report_type_selector.val();
+                dataDownloadObj.$selection_informations.each(function(index, ele) {
+                    if ($(ele).hasClass(selectedOption)) {
+                        $(ele).show();
+                    } else {
+                        $(ele).hide();
+                    }
+                });
+                dataDownloadObj.clear_display();
+            });
+
+            this.selectedOption = function() {
+                return dataDownloadObj.$selectedSection.find('select').find('option:selected');
+            };
+
+            /**
+             * On click download button get selected option and pass it to handler function.
+             */
+            this.downloadReportClickHandler = function() {
+                var selectedOption = dataDownloadObj.selectedOption();
+                var errorMessage = dataDownloadObj.ERROR_MESSAGES[selectedOption.val()];
+
+                if (selectedOption.data('directdownload')) {
+                    location.href = selectedOption.data('endpoint') + '?csv=true';
+                } else if (selectedOption.data('datatable')) {
+                    dataDownloadObj.renderDataTable(selectedOption);
+                } else {
+                    dataDownloadObj.downloadCSV(selectedOption, errorMessage, false);
+                }
+            };
+            this.$download_report.click(dataDownloadObj.downloadReportClickHandler);
+
+            /**
+             * Call data endpoint and invoke buildDataTable to render Table UI.
+             * @param selected option element from report selector to get data-endpoint.
+             * @param errorMessage Error message in case endpoint call fail.
+             */
+            this.renderDataTable = function(selected, errorMessage) {
+                var url = selected.data('endpoint');
+                dataDownloadObj.clear_display();
+                dataDownloadObj.$certificate_display_table.text(gettext('Loading data...'));
```

Comment:
why certificate display table?

Reply:
I will update this name to something more generic.